collective/deletepermission/tests/test_dexterity_delete.py

```diff
@@ -0,0 +1,88 @@
+from collective.deletepermission import testing
+from ftw.builder import Builder
+from ftw.builder import create
+from plone.app.testing import TEST_USER_ID
+from plone.app.testing import TEST_USER_NAME
+from plone.app.testing import login
+from plone.app.testing import setRoles
+from unittest2 import TestCase
+from zExceptions import Unauthorized
+from plone.autoform.interfaces import IFormFieldProvider
+from plone.dexterity.fti import DexterityFTI
+from zope.interface import alsoProvides
+from zope.interface import Interface
+from zope.interface import implements
+from zope import schema
+
+import plone.dexterity.content
+
+
+class IDXFolder(Interface):
+    title = schema.TextLine(
+        title=u'Title',
+        required=False)
+
+alsoProvides(IDXFolder, IFormFieldProvider)
+
+
+class DXFolder(plone.dexterity.content.Container):
+    implements(IDXFolder)
+
+
+class TestDXDeleting(TestCase):
+    layer = testing.COLLECTIVE_DELETEPERMISSION_DX_INTEGRATION_TESTING
+
+    def setUp(self):
+        setRoles(self.layer['portal'], TEST_USER_ID, ['Contributor'])
+        login(self.layer['portal'], TEST_USER_NAME)
+        # add sample fti
+        self.portal = self.layer['portal']
+        self.fti = DexterityFTI('dexterity.folder')
+        self.fti.klass = 'collective.deletepermission.tests.test_dexterity_delete.DXFolder'
+        self.fti.schema = 'collective.deletepermission.tests.test_dexterity_delete.IDXFolder'
+        self.fti.allowed_content_types = ('dexterity.folder',)
+        self.fti.behaviors = ('plone.app.dexterity.behaviors.metadata.IBasic',)
```

Comment:
How about moving creating the dexterity FTI into a Generic Setup profile located in the `tests` folder?
The profile could then be installed by the dexterity testing layer.
We could also go as far as installing it in the normal testing layer and drop the dexterity testing layer; dexterity is importable anyway since it is a testing dependency and the patch will apply, thus it gives us not that much value to have a separate dexterity layer.

Moving FTI creation to a profile would make it simpler here and the FTI could be reused for other tests, such as cut / paste.

Also: `setUp` is called per test, I see this more as a layer setup thing :wink: